tests/test_initialization.py

```diff
@@ -2661,7 +2661,7 @@ def fn(x, *args):
         if prepare_layer_inputs_keys is None:
             prepare_layer_inputs_fn = fn
         else:
-            prepare_layer_inputs_fn = {k: fn for k in prepare_layer_inputs_keys}
+            prepare_layer_inputs_fn = dict.fromkeys(prepare_layer_inputs_keys, fn)
```

Comment:
Please revert, this ruff rule is excluded in the `pyproject.toml` so should not be changed by ruff.